File: solutions/0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> map;
        for (int idx = 0; idx < nums.size(); idx++) {
            int curr_val = nums.at(idx);
            if (map.find(target - curr_val) != map.end()) {
                return vector<int>{map.find(target - curr_val)->second, idx};
            } else {
                map.insert({curr_val, idx});
            }
        }
        return vector<int>{-1, -1};
    }
};

File: solutions/0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # loop through the list and create a dictionary that maps the value to a tuple of its idx and the target - the key value
        targetMap = {}
        for idx, val in enumerate(nums):
            if (targetMap.get(target - val, -1) != -1):
                return [targetMap.get(target - val), idx]
            targetMap.setdefault(val, idx)
            
        
        # for i in range(len(nums)):
        #     for j in range(i + 1, len(nums)):
        #         if (nums[i] + nums[j] == target):
        #             return [i, j]
        # return [-1, -1]
        
        
            
        
        
